### PR TITLE
[DPE-2909] Enable audit logging

### PR DESCRIPTION
**Pull request by dmitry-ratushnyy**
## Issue
In order to improve system security we need to enable audit logging for MongoDB

## Solution
Add audit logging configuration parameters

### CODE REVIEW

File: lib/charms/mongodb/v1/helpers.py [hunk 1]
```diff
@@ -143,6 +143,9 @@ def get_mongod_args(
         # for simplicity we run the mongod daemon on shards, configsvrs, and replicas on the same
         # port
         f"--port={Config.MONGODB_PORT}",
+        "--auditDestination file",
```
Comment (delgod) [inline lib/charms/mongodb/v1/helpers.py]: lets be consistent with delimiter for all options
```suggestion
        "--auditDestination=file",
```

**Review by delgod — CHANGES_REQUESTED** [review]

**Review by MiaAltieri — DISMISSED** [review]
Looks good to me Some questions:
1. how did you test this? 
2. I assume tests are coming in the next PR?
3. Are these changes also necessary for 5/edge branch?

**Comment by dmitry-ratushnyy** [comment]
> Looks good to me Some questions:
> 
> 1. how did you test this?
There are unit tests.
> 2. I assume tests are coming in the next PR?
I think I can add integration tests.
> 3. Are these changes also necessary for 5/edge branch?
No, I don't think so, but we can implement it as well

File: lib/charms/mongodb/v1/helpers.py (hunk 1 above)
Comment (dmitry-ratushnyy) [reply lib/charms/mongodb/v1/helpers.py]: Done, thank you

**Review by delgod — DISMISSED** [review]

**Review by delgod — DISMISSED** [review]

File: tests/integration/test_charm.py [hunk 2]
```diff
@@ -314,3 +316,16 @@ def juju_reports_one_primary(unit_messages):
...
+    """Test that audit log was created and contains actual audit data."""
+    leader_unit = await find_unit(ops_test, leader=True)
+    audit_log_snap_path = "/var/snap/charmed-mongodb/common/var/lib/mongodb/audit.json"
+    audit_log = check_output(
+        f"JUJU_MODEL={ops_test.model_full_name} juju ssh {leader_unit.name} 'sudo cat {audit_log_snap_path}'",
+        stderr=subprocess.PIPE,
+        shell=True,
+        universal_newlines=True,
+    )
+    assert len(audit_log) > 0
```
Comment (MiaAltieri) [inline tests/integration/test_charm.py]: what contents do we expect in the audit log

Comment (dmitry-ratushnyy) [reply tests/integration/test_charm.py]: each entry is a JSON with operation information: timestamp, operation, user, db, etc

File: tests/integration/test_charm.py [hunk 3]
```diff
@@ -265,6 +270,7 @@ async def test_no_password_change_on_invalid_password(ops_test: OpsTest) -> None
     assert password1 == password2
 
 
+@pytest.mark.skip("skip")
```
Comment (MiaAltieri) [inline tests/integration/test_charm.py]: ?

File: tests/integration/test_charm.py [hunk 4]
```diff
@@ -314,3 +320,31 @@ def juju_reports_one_primary(unit_messages):
 
     # cleanup, remove killed unit
     await ops_test.model.destroy_unit(target_unit)
+
+
+async def test_audit_log(ops_test: OpsTest) -> None:
```
Comment (MiaAltieri) [inline tests/integration/test_charm.py:322]: will there be other audit log tests in the future ? if so lets make this its own file

File: tests/integration/test_charm.py [hunk 5]
```diff
@@ -314,3 +320,31 @@ def juju_reports_one_primary(unit_messages):
...
+
+    for line in audit_log.splitlines():
+        if not len(line):
+            continue
+        item = json.loads(line)
+        # basic sanity check
+        assert aduit_log_line_sanity_check(item), "Audit sanity log check failed for first line"
+
+
+def aduit_log_line_sanity_check(entry) -> bool:
```
Comment (MiaAltieri) [inline tests/integration/test_charm.py]: Please add this to helpers file

File: tests/integration/test_charm.py (hunk 3 above)
Comment (dmitry-ratushnyy) [reply tests/integration/test_charm.py]: fixed 😄

File: tests/integration/test_charm.py (hunk 4 above)
Comment (dmitry-ratushnyy) [reply tests/integration/test_charm.py:322]: I don't know, I prefer to keep it here for this PR

File: tests/integration/test_charm.py (hunk 5 above)
Comment (dmitry-ratushnyy) [reply tests/integration/test_charm.py]: Done

**Review by MiaAltieri — DISMISSED** [review]

**Review by MiaAltieri — APPROVED** [review]

**Review by delgod — APPROVED** [review]

**Pull request merged**